metrics/serve.go

```diff
@@ -69,15 +70,12 @@ func RunMetrics(cfg Config, stop chan struct{}) error {
 	}()
 
 	go func() {
-		for tick := range metricTick.C {
-			fmt.Printf("%v: refreshing metric cycle\n", tick)
+		for tick := range cycleTick.C {
+			log.Infof("%v: refreshing cycle", tick)
 			metricsMux.Lock()
-			defer metricsMux.Unlock()
-
-			unregisterMetrics()
+			deleteValues()
 			cycleId++
-			registerMetrics(cfg, cycleId)
-			cycleValues()
+			cycleValues(cycleId)
 			select {
```

Comment:
Don't. you need unlock here?
It seems like you are using metricsMux to protect cycleId and metrics. Won't it be better if you encapsulated the lock/unlock within the functions that directly reference metrics - such as, cycleValues? And there you can use defer to ensure that mutex is unlocked even if exception occurs in those functions/methods.

Reply:
Had fixed the unlock but forgotten to push. It is there now.

Ya that would work too. Although, we would have to keep a mutex in each metric object and we are making thousands of those so not sure if its worth it.

Reply:
Moved inside cycle/deleteValues methods.